packages/x-components/src/x-modules/history-queries/components/my-history.vue

```diff
@@ -82,6 +102,70 @@
      */
     @State('historyQueries', 'historyQueries')
     public historyQueries!: HistoryQuery[];
+
+    /**
+     * It injects {@link SnippetConfig} provided by an ancestor as snippetConfig.
+     *
+     * @internal
+     */
+    @Inject('snippetConfig')
+    public snippetConfig!: SnippetConfig;
+
+    /**
+     * Returns history queries group by date.
+     *
+     * @example
+     * ```typescript
+     *  const historyQueriesGrouped = \{
+     *    "Monday, January 10th, 2022" : [
+     *      \{
+     *        query: 'lego',
+     *        modelName: 'HistoryQuery',
+     *        timestamp: 121312312
+     *      \}
+     *    ],
+     *     "Tuesday, January 11th, 2022" : [
+     *      \{
+     *        query: 'barbie',
+     *        modelName: 'HistoryQuery',
+     *        timestamp: 15221212
+     *      \}
+     *    ]
+     *  \}
+     * ```
+     *
+     * @returns HistoryQueries - The history queries grouped by date.
+     *
+     * @internal
+     */
+    public get historyQueriesGroupByDate(): Record<string, HistoryQuery[]> {
+      return groupItemsBy(this.historyQueries, current => {
+        return new Date(current.timestamp).toLocaleDateString(this.snippetConfig.lang, {
+          day: 'numeric',
+          weekday: 'long',
+          month: 'long',
+          year: 'numeric'
+        });
+      });
+    }
```

Comment:
```suggestion
    /**
     * The `groupByDate` property is a computed property that returns a record of history queries
     * grouped by date.
     *
     * @example
     * ```typescript
     *  const historyQueriesGrouped = {
     *    'Monday, January 10th, 2022' : [{
     *      query: 'lego',
     *      modelName: 'HistoryQuery',
     *      timestamp: 121312312
     *    }],
     *    'Tuesday, January 11th, 2022' : [{
     *      query: 'barbie',
     *      modelName: 'HistoryQuery',
     *      timestamp: 15221212
     *    }]
     *  }
     * ```
     *
     * @returns The history queries grouped by date.
     *
     * @internal
     */
    protected get groupByDate(): Dictionary<HistoryQuery[]> {
      return groupItemsBy(this.historyQueries, current => {
        return new Date(current.timestamp).toLocaleDateString(this.snippetConfig.lang, {
          day: 'numeric',
          weekday: 'long',
          month: 'long',
          year: 'numeric'
        });
      });
    }
```